### PR TITLE
tests: kernel: fpu_sharing: move to new ztest API

### PR DESCRIPTION
**Pull request by ghost**
Move tests/kernel/fpu_sharing/ to use new ztest API.
And increase main stack size for qemu_leon3.

Signed-off-by: Guo Lixin [lixinx.guo@intel.com](mailto:lixinx.guo@intel.com)

### CODE REVIEW

**Review by dcpleung — DISMISSED** [review]

**Comment by LixinGuoX** [comment]
CI blocked on qemu_leon3(see #47477), so increase the main stack size for qemu_leon3 as a workaround.

File: tests/kernel/fpu_sharing/float_disable/src/k_float_disable.c [hunk 1]
```diff
@@ -263,9 +263,11 @@ void test_k_float_disable_irq(void)
 	zassert_true(ok, "");
 }
 #else
-void test_k_float_disable_irq(void)
+ZTEST(k_float_disable, test_k_float_disable_irq)
```
Comment (enjiamai) [inline tests/kernel/fpu_sharing/float_disable/src/k_float_disable.c:266]: Personal suggestion: Put the #if / #else to info function definition of k_float_disable,  you don't have to write ZTEST(k_float_disable,....)  twice.

Comment (LixinGuoX) [reply tests/kernel/fpu_sharing/float_disable/src/k_float_disable.c:266]: For this #if /#else, we have some #include and other functions inside, those are not suitable to put into function test_k_float_disable_irq

Comment (LixinGuoX) [reply tests/kernel/fpu_sharing/float_disable/src/k_float_disable.c:266]: > Personal suggestion: Put the #if / #else to info function definition of k_float_disable, you don't have to write ZTEST(k_float_disable,....) twice.

For this #if /#else, we have some #include and other functions inside, those are not suitable to put into function test_k_float_disable_irq

Comment (enjiamai) [reply tests/kernel/fpu_sharing/float_disable/src/k_float_disable.c:266]: Hi @yperess , may I ask you a question? Any suggestion of  "should we keep one ZTEST function definition as possible or it's ok for both"? Thanks!

**Review by enjiamai — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Pull request merged**